### DISCUSSION TITLE
Support to have a Kanister function that can run another Kanister function for `n` resources

### DISCUSSION THREAD

**Discussion by viveksinghggits** [kanisterio/kanister#3001] [Ideas]
Like discussed here https://kanisterio.slack.com/archives/C85C8V22J/p1721938464102899, it would be a really great enhancement if we can have a Kanister function executed for `n` resources without actually having those Kanister function repeated in `n` phases.

It would be really helpful for the cases where we want to, let's say 
- scale down 4 deployments
- take CSI snapshot of 5 PVCs

Since we already have function to scale down a workload and take csi snapshot of a PVC there should be a way to do that for `n` times or `n` resources.

**Comment by hairyhum** [comment]
I'm curious how would such function handle function arguments and which inputs will need to be changed?
We have
- objects coming form the actionset
- profile 
- function arguments

Which ones would need to be dynamic in that case?

Also making an "iterate" function is one approach, but maybe we can make something in the actionset to iterate phases. Maybe we can make some examples of what the interface could look like?

**Comment by hairyhum** [comment]
https://github.com/kanisterio/kanister/pull/3414
